service.py

```diff
@@ -187,7 +189,16 @@ def calculate_metric(article):
         dislikes = article.downvote
         visits = article.visit
         comments = Comment.query.filter_by(article_id=article.id).count()
-        metric = likes * 50 - dislikes * 30 + visits * 10 + comments * 20
+        # positive feedback
+        visits_score = math.log10(visits) * 2
+        comments_score = comments * 5
+        likes_score = (likes - dislikes) * 3
```

Comment:
@CNZedChou 

Let's say the lowest possible value for popularity is 0.   Can `likes_score = max(likes - dislikes, 0) * 3` be a better formula for computing `likes_score`?

-Hui